tactics/eauto.ml

```diff
@@ -41,7 +41,12 @@ let assumption id = e_give_exact (mkVar id)
 
 let e_assumption =
   Proofview.Goal.enter { enter = begin fun gl ->
-    Tacticals.New.tclFIRST (List.map assumption (Tacmach.New.pf_ids_of_hyps gl))
+    (if Flags.version_strictly_greater Flags.V8_6 then
```

Comment:
That's not entirely true: `assumption` on a « dependent goal » it will affect the other goals. Illustration:

```coq
Goal exists x, x = true.
Proof.
  set (t := true).
  set (f := false).
  pose proof (eq_refl true) as h.
  unshelve refine (ex_intro _ _ _).
  Fail all:assumption.
  all: move t after f; assumption.
Qed.
```